### ISSUE TITLE
iOS Crash: Documentation or Code clarification required

### ISSUE THREAD

**Issue by barocsi** [BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK#131]
This function crashes when calling generateShortURL
```
- (void)generateShortUrl:(CDVInvokedUrlCommand*)command
{
    NSLog(@"start generateShortUrl");

    int branchUniversalObjectId = [[command.arguments objectAtIndex:0] intValue];
    NSDictionary *arg1 = [command.arguments objectAtIndex:1];
    NSDictionary *arg2 = [command.arguments objectAtIndex:2];

    BranchUniversalObject *branchUniversalObj = [self.branchUniversalObjArray objectAtIndex:branchUniversalObjectId];

    BranchLinkProperties *props = [[BranchLinkProperties alloc] init];

    for (id key in arg1) {
        if ([key isEqualToString:@"duration"]) {
            props.matchDuration = (NSUInteger)[((NSNumber *)[arg1 objectForKey:key]) integerValue];
        }
        else {
            [props setValue:[arg1 objectForKey:key] forKey:key];
        }
    }

    for (id key in arg2) {
        [props addControlParam:key withValue:[arg1 objectForKey:key]];
    }

    [branchUniversalObj getShortUrlWithLinkProperties:props andCallback:^(NSString *url, NSError *error) {
        NSLog(@"inside getShortUrlWithLinkProperties block");
        CDVPluginResult* pluginResult = nil;
        if (!error) {
            NSError *err;
            NSDictionary *jsonObj = [[NSDictionary alloc] initWithObjectsAndKeys:url, @"url", 0, @"options", &err, @"error", nil];

            if (!jsonObj) {
                NSLog(@"Parsing Error: %@", [err localizedDescription]);
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:[err localizedDescription]];
            } else {
                NSLog(@"Success");
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:jsonObj];
            }
        }
        else {
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:[error localizedDescription]];
        }
        NSLog(@"returning data to js interface..");
        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    }];
}
```
because 
```
 for (id key in arg2) {
        [props addControlParam:key withValue:[arg1 objectForKey:key]];
    }
```
because wen calling

```
branchUniversalObj.generateShortUrl({
                                      "feature": "invite",
                                      "channel": "Facebook",
                                      "stage": stage
                                    },{}).then
```
the {} is mandatory, either null or empty. Without it the app crashes due to the reason above. Fix either this or documentation. If the plugin exposes something that is mishandled, it should not crash the app itself. The put everything into try catch within the native part so it will not kill the app.

**Comment by aaustin** [comment]
Gotcha in this commit to my big PR: https://github.com/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/pull/120/commits/6eacba442e9d4adf38d9b826f48b1d887cd895bf

**Comment by aaustin** [comment]
I fixed it in 2.0.12
